### DISCUSSION TITLE
[V3] BadMethodCallException: Method Illuminate\Http\Response::assertNoRedirect does not exist.

### DISCUSSION THREAD

**Discussion by faltjo** [livewire/livewire#6261] [7. Bugs (archived)] [answered]
### Livewire version

3.0.0-beta.8

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/faltjo/livewire-v3-assert-no-redirect

### Steps To Reproduce

1. Create a [component](https://github.com/faltjo/livewire-v3-assert-no-redirect/blob/main/app/Livewire/CreatePost.php) that does not redirect.
2. Write a `[test](https://github.com/faltjo/livewire-v3-assert-no-redirect/blob/main/tests/Feature/Livewire/CreatePostTest.php)` that asserts that the livewire component does not redirect.
3. Run the test, e.g. with `./vendor/bin/phpunit tests/Feature/Livewire`

It fails with 

```
There was 1 error:

1) Tests\Feature\Livewire\CreatePostTest::does_not_redirect
BadMethodCallException: Method Illuminate\Http\Response::assertNoRedirect does not exist.

livewire-v3-assert-no-redirect/vendor/laravel/framework/src/Illuminate/Macroable/Traits/Macroable.php:112
livewire-v3-assert-no-redirect/vendor/laravel/framework/src/Illuminate/Testing/TestResponse.php:1756
livewire-v3-assert-no-redirect/vendor/livewire/livewire/src/Features/SupportRedirects/TestsRedirects.php:38
livewire-v3-assert-no-redirect/tests/Feature/Livewire/CreatePostTest.php:17
livewire-v3-assert-no-redirect/vendor/laravel/framework/src/Illuminate/Foundation/Testing/TestCase.php:174

```

### Additional information

Test works when using `"livewire/livewire": "^2.0"`.
`assertNoRedirect` is still available in Livewire 3, according to https://livewire.laravel.com/docs/testing#assertions

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by sinnbeck** [comment]
Couldnt you normally call a method first? It seems to work for me if I do that.

```php
    public function does_not_redirect()
    {
        Livewire::test(CreatePost::class)
           ->call('save')
            ->assertNoRedirect()
            ->assertStatus(200);
    }

**Reply by faltjo** [reply]
Indeed, if I call a method first, the `assertNoRedirect` works. But shouldn't it work also for simple components, where there is just rendering?

I mean just using `assertStatus` for example (see below) does also work, right?
```
public function is_successful()
{
    Livewire::test(CreatePost::class)
        ->assertStatus(200);
}
    
```

**Reply by sinnbeck** [reply]
The problem is that it passes the method to laravel, which does not have a `assertNoRedirect()` method. This happens as it isn't a livewire request. https://github.com/livewire/livewire/blob/main/src/Features/SupportRedirects/TestsRedirects.php#L37-L41

https://laravel.com/docs/10.x/http-tests#response-assertions

So yes there is a bug, as it tries to call a non-existent method, but I am unsure how its supposed to be fixed. Probably the marked code should be removed again in livewire.

**Reply by sinnbeck** [reply]
Just tested and it indeed works if I remove that code block from livewire. A PR for this might be a good idea :)

**Reply by faltjo** [reply]
Thanks for elaborating @sinnbeck. I created a PR with the suggested changes (https://github.com/livewire/livewire/pull/6288), although I have to admit, that I am not sure, why the affected lines were added, see https://github.com/livewire/livewire/commit/62a909b8a0dccc7089e258caa3340c55dce563b2.

**Reply by faltjo** [reply, marked as answer]
Fixed by https://github.com/livewire/livewire/pull/6288
